server/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -173,45 +169,23 @@ public DocTableInfo create(RelationName relationName, Metadata metadata) {
         return docTableInfoFactory.create(relationName, metadata);
     }
 
+    @Override
+    public DocTableInfo create(IndexMetadata indexMetadata) {
+        return docTableInfoFactory.create(indexMetadata);
+    }
+
     private static long getTableVersion(Metadata metadata, RelationName name) {
         RelationMetadata relation = metadata.getRelation(name);
         if (relation instanceof RelationMetadata.Table table) {
             return table.tableVersion();
         }
```

Comment:
```suggestion
        RelationMetadata.Table table = metadata.getRelation(name);
        return table.tableVersion();
```

Can rely on the cast logic within `getRelation` to avoid the unreachable case below